src/main/java/Command.java

```diff
@@ -0,0 +1,36 @@
+import java.io.IOException;
+
+/**
+ * The abstract Command class is used for extension to the different commands for different interaction
+ * with the saved TaskList and output printed to the user.
+ */
+abstract class Command {
+    protected String command;
+    protected TaskList tasks;
+    protected Ui ui;
+    protected Storage storage;
+
+    /**
+     * This method is used to terminate the run() method in the Duke class.
+     *
+     * @return This returns true if the command is an ExitCommand and false otherwise.
+     */
+    boolean isExit() {
+        if (this instanceof ExitCommand) {
```

Comment:
Should avoid using this line, as now there is dual-dependency (ie. Command depends on ExitCommand and ExitCommand also depends on Command)